internal/cli/atlas_logs_download.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"io"
+	"os"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flags"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type atlasLogsDownloadOpts struct {
+	globalOpts
+	host  string
+	name  string
+	out   string
+	start string
+	end   string
+	fs    afero.Fs
+	store store.LogsDownloader
+}
+
+func (opts *atlasLogsDownloadOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasLogsDownloadOpts) Run() error {
+	f, err := opts.newWriteCloser()
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	r := opts.newDateRangeOpts()
+	if err := opts.store.DownloadLog(opts.ProjectID(), opts.host, opts.name, f, r); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (opts *atlasLogsDownloadOpts) output() string {
+	if opts.out == "" {
+		opts.out = opts.name
+	}
+	return opts.out
+}
+
+func (opts *atlasLogsDownloadOpts) newWriteCloser() (io.WriteCloser, error) {
+	ff := os.O_CREATE | os.O_TRUNC | os.O_WRONLY | os.O_EXCL
```

Comment:
😮 👍

Reply:
Added a comment to this as it's a bit mystical